### PR TITLE
chore: fix code documentation on TypeVar variables that make Sphinx build fails (#58)

### LINKED ISSUE #58: Sphinx ^= 6.1.0 couldn't build our documentation
**Description**
In the latest version of Sphinx (from [6.1.0](https://www.sphinx-doc.org/en/master/changes.html#release-6-1-0-released-jan-05-2023) to [6.1.3.](https://www.sphinx-doc.org/en/master/changes.html#release-6-1-3-released-jan-10-2023)), there is a breaking change that prevent the Macaron's documentation from being built successfully.
When building the documentation with `make clean docs`, there following error messages is seen (some part of the message is removed for visibility):
```
Running Sphinx v6.1.0
making output directory... done
[autosummary] generating autosummary for: "***"
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 18 source files that are out of date
updating environment: [new config] 18 added, 0 changed, 0 removed
ERROR: [numpydoc] While processing docstring for 'macaron.code_analyzer.call_graph.Node'                                                                                                            

Extension error (numpydoc.numpydoc):
Handler <function mangle_docstrings at 0x7fbe18f79d00> for event 'autodoc-process-docstring' threw an exception (exception: Expected a class or None, but got ~Node)
make[1]: *** [Makefile:23: html] Error 2
make[1]: Leaving directory '***'
make: *** [Makefile:230: docs/_build/html/index.html] Error 2
```
The message indicates that the error happened for the global variable [here](https://github.com/oracle-samples/macaron/blob/babd2e784fcea81ed6b383ad3ad992f9d6ab8cc7/src/macaron/code_analyzer/call_graph.py#L9).

**Solution**
- We could revert the version of Sphinx back to 6.0.0 so that the error won't happen.

### PR DESCRIPTION
**Pull request by tromai**
Close #58.

### CODE REVIEW

File: src/macaron/code_analyzer/call_graph.py [hunk 1]
```diff
@@ -7,11 +7,14 @@
 from typing import Generic, TypeVar
 
 Node = TypeVar("Node", bound="BaseNode")
-"""This binds type ``Node`` to ``BaseNode`` and any of its subclasses.
-
-Therefore, any node of type ``Node`` that is stored in the call graph
-container will be a subtype of ``BaseNode``.
-"""
+# This code documentation is commented out because newer Sphinx version (^=6.1.0) has issues processing
```
Comment (behnazh-w) [inline src/macaron/code_analyzer/call_graph.py:10]: Have you tried adding `sphinx_toolbox.more_autodoc.typevars` to `conf.py`?
See https://sphinx-toolbox.readthedocs.io/en/latest/extensions/more_autodoc/typevars.html

Comment (tromai) [reply src/macaron/code_analyzer/call_graph.py:10]: I have tried but I still received the same error.

**Comment by tromai** [comment]
https://github.com/sphinx-doc/sphinx/issues/10700 Might be relevant to the issue we are having.

File: src/macaron/code_analyzer/call_graph.py [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
+# it.
```
Comment (behnazh-w) [inline src/macaron/code_analyzer/call_graph.py:11]:
```suggestion
# The documentation below for `TypeVar` is commented out due to a breaking 
# change in Sphinx version (^=6.1.0).
```

File: src/macaron/output_reporter/results.py [hunk 3]
```diff
@@ -62,7 +62,10 @@ class DepSummary(TypedDict):
 
 
 RecordNode = TypeVar("RecordNode", bound="Record")
-"""This binds type ``RecordNode`` to ``Record`` and any of its subclasses."""
+# This code documentation is commented out because newer Sphinx version (^=6.1.0) has issues processing
+# it.
```
Comment (behnazh-w) [inline src/macaron/output_reporter/results.py:66]:
```suggestion
# The documentation below for `TypeVar` is commented out due to a breaking 
# change in Sphinx version (^=6.1.0).
```

File: src/macaron/code_analyzer/call_graph.py (hunk 2 above)
Comment (tromai) [reply src/macaron/code_analyzer/call_graph.py]: Fix in https://github.com/oracle-samples/macaron/pull/59/commits/5551cbc784af5e36339580c7a1851f5e92cbb109

File: src/macaron/output_reporter/results.py (hunk 3 above)
Comment (tromai) [reply src/macaron/output_reporter/results.py]: Fix in https://github.com/oracle-samples/macaron/pull/59/commits/5551cbc784af5e36339580c7a1851f5e92cbb109

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**